src/clockwork/amqp.clj

```diff
@@ -26,8 +37,26 @@
 
 (defn connect
   [exchange-cfg queue-cfg handlers]
-  (let [channel (lch/open (rmq/connect {:uri (config/amqp-uri)}))]
+  (let [connection (rmq/connect {:uri (config/amqp-uri)})
+        channel (lch/open connection)]
     (log/info (format "[amqp/connect] [%s]" (config/amqp-uri)))
     (declare-exchange channel exchange-cfg)
     (declare-queue channel exchange-cfg queue-cfg (keys handlers))
-    (lc/blocking-subscribe channel (:name queue-cfg) (partial message-router handlers))))
+    (lc/blocking-subscribe channel (:name queue-cfg) (partial message-router handlers))
+    connection))
+
+(defn publish-msg
+  [routing-key msg]
+  (try+
+    (let [timeNow (new java.util.Date)
+          channel (lch/open (connection))]
```

Comment:
Continuing to be a bit fiddly! We should probably close the channel at the outset of this function (just `(close channel)`)